src/core/service.c

```diff
@@ -3047,6 +3087,15 @@ static void service_notify_message(Unit *u, pid_t pid, char **tags, FDSet *fds)
                 service_add_fd_store_set(s, fds, name);
         }
 
+        e = strv_find_startswith(tags, "WATCHDOG_USEC=");
+        if (e) {
+                usec_t watchdog_override_usec;
+                if (parse_time(e, &watchdog_override_usec, 1) < 0)
```

Comment:
parse_time() will actually parse expressions like "5h 3min", i.e. is for user-facing time specifications. In this case the message is coming from a program however, and I think we should be stricter here, and really only parse usec specifications, the same way as sd_watchdog_enabled() does it. Or in other words, please use safe_atou64() here instead of parse_time().